Model/DutyCalculator.php

```diff
@@ -116,84 +116,333 @@ public function __construct(
         $this->_logger = $logger;
     }
 
+
+    /**
+     * Sets quote and checkout Session state based on whether duty and tax is applicable or not
+     * @param Magento\Quote\Api\Data\AddressInterface $address
+     * @param bool $apply  - whether duty is applicable or not
+     *
+     */
+    public function handleTaxApplicability($address, $apply)
+    {
+        $this->_logger->debug('In handleTaxApplicability method');
+        $this->_logger->debug($this->checkoutSession->getReachDuty());
+        $this->_logger->debug($address->getCountryId());
+        $this->_logger->debug($address->getRegionCode());
+        $this->_logger->debug($this->checkoutSession->getPrevCountry());
+        $this->_logger->debug($this->checkoutSession->getPrevRegion());
+        $this->_logger->debug($apply);
+        $quote = $this->checkoutSession->getQuote();
+
+        if ($apply || !$this->getIsOptional($address->getCountryId())) {
+           $quote->setReachDuty($this->checkoutSession->getReachDuty());
+           $this->response->setIsOptional($this->getIsOptional($address->getCountryId()));
+           $this->response->setSuccess(true);
+           $this->response->setDuty($this->checkoutSession->getRechDuty());
+           $this->_logger->debug('In handleTaxApplicability method --- apply is true');
```

Comment:
this debug comment is a bit misleading, right? Because either apply is true or no CountryId could be retrieved, the code could still run even if apply was false in this case

Reply:
Right. During debug message beautification (replacing old one before commit) this happened. :( Going to change.

Reply:
of course! totally understand, thank you!

Reply:
https://github.com/withreach/magento-plugin/pull/44/commits/43e8c0cb06c299c3f0cd941e2deecbe32d2c39de should handle it.  
Remove the debug comment altogether though. Thanks.

Reply:
Adding more useful debug log (hopefully) for this section as I saw your other/newer statement by now. Thanks.

Reply:
This commit 87a79f49461544e2dea9c01befe0a2a40ff34b0c has different/updated log messages now. Please let me know if that is not clear enough/should be changed too. Thanks.